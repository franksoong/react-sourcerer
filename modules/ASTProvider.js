import React from 'react';
import {parse} from 'babylon';

class ASTProvider extends React.Component {
  getChildContext() {
    return {
      ast: parse(this.props.source, {
        sourceType: 'module',
        plugins: [
          'jsx',
        ],
      }),
    };
  }

  render() {
    return (
      React.Children.only(this.props.children)
    );
  }
}

ASTProvider.childContextTypes = {
  ast: React.PropTypes.object,
};

export default ASTProvider;
