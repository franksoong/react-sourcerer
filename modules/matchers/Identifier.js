import React from 'react';
import Match from '../Match';
import * as t from 'babel-types';

class Identifier extends React.Component {
  render() {
    return (
      <Match
        {...this.props}
        match={(path) => {
          return t.isIdentifier(path);
        }}
      />
    );
  }
}

export default Identifier;
