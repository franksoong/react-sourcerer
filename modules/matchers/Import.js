import React from 'react';
import Match from '../Match';
import * as t from 'babel-types';

class Import extends React.Component {
  render() {
    return (
      <Match
        {...this.props}
        match={(path) => {
          const {from} = this.props;
          const isImportDeclaration = t.isImportDeclaration(path);
          if (!isImportDeclaration) return false;
          if (from && from !== path.node.source.value) return false;
          return isImportDeclaration;
        }}
      />
    );
  }
}

export default Import;
