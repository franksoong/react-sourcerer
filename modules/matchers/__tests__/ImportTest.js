import ASTProvider from '../../ASTProvider';
import React from 'react';
import Identifier from '../Identifier';
import Import, { getIdentifiersFromImport } from '../Import';

import renderer from 'react-test-renderer';

describe('Import', () => {
  it('Should sub match for identifier', () => {
    const output = renderer.create(<ASTProvider source={`
    import _, { forEach } from 'lodash';
    `}>
    <Import from={'lodash'}>
      {(imports) => (
        <div>
          {imports.map((match, index) => (
            <Identifier ast={match} key={index}>
              {(identifiers) => (
                <div>
                  {identifiers.map((identifier, n) => 
                    <div key={n}>{identifier.node.name}</div>
                  )}
                </div>
              )}
            </Identifier>
          ))}
        </div>
      )}
    </Import>
  </ASTProvider>);

    expect(output.toJSON()).toMatchSnapshot(); 
  });

  it('should extract all locals from import', () => {
    const output = renderer.create(<ASTProvider source={`
    import _, { forEach } from 'lodash';
    `}>
    <Import from={'lodash'}>
      {(imports) => (
        <div>{getIdentifiersFromImport(imports)}</div>
      )}
    </Import>
  </ASTProvider>);

    expect(output.toJSON()).toMatchSnapshot(); 
  });
});
