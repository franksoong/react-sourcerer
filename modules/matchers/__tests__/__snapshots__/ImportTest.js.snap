exports[`Import Should sub match for identifier 1`] = `
<div>
  <div>
    <div>
      _
    </div>
    <div>
      forEach
    </div>
    <div>
      forEach
    </div>
  </div>
</div>
`;
